core/test/test_optimizers.cpp

```diff
@@ -0,0 +1,85 @@
+#include <catch.hpp>
+#include <Spirit/State.h>
+#include <Spirit/Configurations.h>
+#include <Spirit/Parameters.h>
+#include <Spirit/Geometry.h>
+#include <Spirit/Simulation.h>
+#include <Spirit/System.h>
+#include <Spirit/Quantities.h>
+
+TEST_CASE( "Optimizers testing", "[optimizers]" )
+{
+  // input File
+  const char * inputfile = "core/test/input/optimizers/test.cfg";
+
+  // optimizers and outputs
+  std::vector<const char *>  optimizers { "SIB", 
+                                          "SIB2", 
+                                          "VP",
+                                          "Heun" };   // optimizers to be tested
+  int index;    // index of optimizer in optimizers list  
+  std::vector<float>        energy( optimizers.size(), .0 );      // energy output
+  std::vector<float [3]>    magnetization( optimizers.size() );   // magnetization output
```

Comment:
Use `Engine::Vectormath::Magnetization` instead.
In the core tests you should not use the API functions so much.